airspeed/__init__.py

```diff
@@ -628,14 +628,26 @@ class SubExpression(_Element):
     DOT = re.compile(r'\.(.*)', re.S)
 
     def parse(self):
-        self.identity_match(self.DOT)
-        self.expression = self.next_element(VariableExpression)
+        try:
```

Comment:
Isn't `SubExpression` for things that look like `$foo.bar`, which I don't think is reflected in the new unit tests? Presumably you found this change was necessary for that case, and that the test cases you added were already working.

Reply:
I guess I'm wondering if the change needs to be even more general, e.g. for Value, so that `[2]` could be added after a parenthesised expression, for example. Not sure what velocity supports these days.

Reply:
@purcell  Thanks for the feedback - it's quite possible that there may be a better / more general approach. (I am fairly new to the code base and not too familiar with all the expressions/rules.)

Would be great if you can provide some guidance. How would you like to proceed - do you want me to extend the tests, or would you prefer to work on an integration with `Value`?

Another option would be to merge the change as-is, and potentially extend the logic with support for `Value`/parenthesised expressions in a future PR (at least now we'd already have this additional functionality covered by unit tests..). Thanks

Reply:
@purcell Any suggestions on how to proceed with this PR? We have already merged this change into [LocalStack](https://github.com/localstack/localstack) where this library is being used (and the change seems to be working well), but would be happy to make and necessary adjustments to get this PR merged here as well. Thanks!

Reply:
Yeah, sorry for the radio silence. I had a look at this at the time but didn't come to a firm conclusion about where best to make the change. I'm happy to merge this PR as-is if you wouldn't mind providing a test case. :-)